src/codecs/dav1d.rs

```diff
@@ -143,7 +143,11 @@ impl Decoder for Dav1d {
                 image.width = dav1d_picture.p.w as u32;
                 image.height = dav1d_picture.p.h as u32;
                 image.depth = dav1d_picture.p.bpc as u8;
-                image.planes2[3] = Some(Pixels::Pointer(dav1d_picture.data[0] as *mut u8));
+                image.planes2[3] = Some(if image.depth > 8 {
```

Comment:
Why Some() here and u16/u8 and not above?

Reply:
this is an Optional type. i.e. the value can be something or nothing. so we have to wrap it with `Some()` when assigning it and unwrap it with `unwrap()` when using it.

for primitive types like u8/u16 we don't need them to be optional since we can use the value 0 to mean nothing.

Reply:
> this is an Optional type. i.e. the value can be something or nothing. so we have to wrap it with `Some()` when assigning it and unwrap it with `unwrap()` when using it.

I think `match optional { Some(x) => ... }` is preferred to `optional.unwrap()` in Rust, but the added verbosity/error catching seems rarely worth it.

> for primitive types like u8/u16 we don't need them to be optional since we can use the value 0 to mean nothing.

Here it is fine but I would use `Option`s when possible, it is clearer than 0.

Reply:
> > this is an Optional type. i.e. the value can be something or nothing. so we have to wrap it with `Some()` when assigning it and unwrap it with `unwrap()` when using it.
> 
> I think `match optional { Some(x) => ... }` is preferred to `optional.unwrap()` in Rust, but the added verbosity/error catching seems rarely worth it.
> 

Right. It does depend on the usage and whether we care about all matches or just one.

> > for primitive types like u8/u16 we don't need them to be optional since we can use the value 0 to mean nothing.
> 
> Here it is fine but I would use `Option`s when possible, it is clearer than 0.

Yep, agreed.